src/frontends/tensorflow_common/src/op/conj.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "helper_ops/complex_type_mark.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_conj_op (const NodeContext& node) {
+     default_op_checks(node, 1, {"Conjugate"}, true);
+
+    auto x = node.get_input(0);
+
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+
+    std::shared_ptr<Node> conj {x.get_node_shared_ptr()};
+    if (complex_type_mark) {
+        element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+        auto x = complex_type_mark->input_value(0);
+
+        auto real_index = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+        auto imag_index = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+
+        auto gather_axis = make_shared<v0::Constant>(element::i32, Shape{1}, -1);
+
+        auto real = make_shared<v8::Gather>(x, real_index, gather_axis)->output(0);
+        auto imag = make_shared<v8::Gather>(x, imag_index, gather_axis)->output(0);
+
+        imag = make_shared<v0::Negative>(imag);
+
+        auto conj = make_shared<v0::Concat>(OutputVector{real, imag}, -1); 
+
+        set_node_name(node.get_name(), conj);
+        auto complex_conj = make_shared<ComplexTypeMark>(conj, complex_part_type);
+        return {complex_conj->output(0)};
+    }
+    
+    set_node_name(node.get_name(), conj);
+    return {conj};
+
+}
```

Comment:
please move this logic into `conj_transpose.cpp` or vice versa and create common helper for both translators because I see much duplicate in the code.

Reply:
> please move this logic into `conj_transpose.cpp` or vice versa and create common helper for both translators because I see much duplicate in the code.

It looks like the operation files don't have headers, I've written a small helper function, but should I then give each class a helper that gets a shared conj ptr? Is there any neat way to create one common helper and call it from each .cpp file?

Reply:
Hi @jvr0123, no need to create header file. Please place this common helper in `conj_transpose.cpp` in nameless space and use it by both translators for `Conj` and `ConjTranspose`.